corehq/apps/linked_domain/applications.py

```diff
@@ -53,6 +53,7 @@ def link_app(linked_app, master_domain, master_id, remote_details=None):
     DomainLink.link_domains(linked_app.domain, master_domain, remote_details)
 
     linked_app.master = master_id
+    linked_app.progenitor_app_id = master_id
```

Comment:
Similar idea here, though I'm less certain of this one.  What if this master app is itself a copy?  Should we support that?